### PR TITLE
olm pkg, csv handler: new methods were added

### PR DESCRIPTION
**Pull request by elenagerman**
IsSucceeded, GetPhase and ListAllInNamespaceWithNamePattern methods added to the olm pkg, csv handler

### CODE REVIEW

**Comment by elenagerman** [comment]
@mcornea @kononovn @trewest PTAL

File: pkg/olm/clusterserviceversion.go [hunk 1]
```diff
@@ -123,6 +123,56 @@ func (builder *ClusterServiceVersionBuilder) GetAlmExamples() (string, error) {
...
+func (builder *ClusterServiceVersionBuilder) IsSucceeded() (bool, error) {
+	if valid, err := builder.validate(); !valid {
+		return false, err
+	}
+
+	glog.V(100).Infof("Verify %s csv in %s namespace is Succeeded",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	csvFound := builder.Exists()
+	if !csvFound {
```
Comment (trewest) [inline pkg/olm/clusterserviceversion.go:136]: Please simplify it
```suggestion
	if !builder.Exists() {
```

File: pkg/olm/clusterserviceversion.go [hunk 2]
```diff
@@ -123,6 +123,56 @@ func (builder *ClusterServiceVersionBuilder) GetAlmExamples() (string, error) {
...
+func (builder *ClusterServiceVersionBuilder) GetPhase() (oplmV1alpha1.ClusterServiceVersionPhase, error) {
+	if valid, err := builder.validate(); !valid {
+		return "", err
+	}
+
+	glog.V(100).Infof("Get %s csv in %s namespace phase",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	cvFound := builder.Exists()
+	if !cvFound {
```
Comment (trewest) [inline pkg/olm/clusterserviceversion.go:166]: Please simplify it
```suggestion
	if !builder.Exists() {
```

File: pkg/olm/clusterserviceversion.go [hunk 3]
```diff
@@ -123,6 +123,56 @@ func (builder *ClusterServiceVersionBuilder) GetAlmExamples() (string, error) {
...
+// IsSucceeded check if the csv is Succeeded.
+func (builder *ClusterServiceVersionBuilder) IsSucceeded() (bool, error) {
+	if valid, err := builder.validate(); !valid {
+		return false, err
+	}
+
+	glog.V(100).Infof("Verify %s csv in %s namespace is Succeeded",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	csvFound := builder.Exists()
```
Comment (trewest) [inline pkg/olm/clusterserviceversion.go:135]: builder.GetPhase() already calls Exists. I think you can remove it from here

File: pkg/olm/clusterserviceversion.go [hunk 4]
```diff
@@ -123,6 +123,49 @@ func (builder *ClusterServiceVersionBuilder) GetAlmExamples() (string, error) {
...
+
+	glog.V(100).Infof("Get %s csv in %s namespace phase",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	if !builder.Exists() {
+		return "", fmt.Errorf("%s csv not found in %s namespace",
+			builder.Definition.Name, builder.Definition.Namespace)
+	}
+
+	phase := builder.Object.Status.Phase
```
Comment (trewest) [inline pkg/olm/clusterserviceversion.go:164]:
```suggestion
	return builder.Object.Status.Phase, nil
```

File: pkg/olm/clusterserviceversion.go [hunk 5]
```diff
@@ -123,6 +123,49 @@ func (builder *ClusterServiceVersionBuilder) GetAlmExamples() (string, error) {
...
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	phase, err := builder.GetPhase()
+
+	if err != nil {
+		return false, fmt.Errorf("failed to get phase value for %s clusterserviceversion in %s namespace due to %w",
+			builder.Definition.Name, builder.Definition.Namespace, err)
+	}
+
+	if phase == "Succeeded" {
```
Comment (trewest) [inline pkg/olm/clusterserviceversion.go:142]: I think you can just return here. IMO error should be reserved for things that go wrong while performing an operation. 
```suggestion
        return phase == "Succeeded", nil
```

File: pkg/olm/clusterserviceversionlist.go [hunk 6]
```diff
@@ -45,3 +46,30 @@ func ListClusterServiceVersion(
 
 	return csvObjects, nil
 }
+
+// ListAllInNamespaceWithNamePattern returns a cluster-wide csv inventory.
+func ListAllInNamespaceWithNamePattern(
```
Comment (trewest) [inline pkg/olm/clusterserviceversionlist.go:51]:
```suggestion
func ListClusterServiceVersionWithNamePattern(
```

File: pkg/olm/clusterserviceversionlist.go [hunk 7]
```diff
@@ -45,3 +46,30 @@ func ListClusterServiceVersion(
...
+	apiClient *clients.Settings,
+	namePattern, nsname string,
+	options metaV1.ListOptions) ([]*ClusterServiceVersionBuilder, error) {
+	glog.V(100).Infof("Listing csv filtered by the name pattern %s are found in %s namespace", namePattern, nsname)
+
```
Comment (trewest) [inline pkg/olm/clusterserviceversionlist.go:56]: Validate that namePattern is not an empty string

**Review by trewest — CHANGES_REQUESTED** [review]
Just a few comments

File: pkg/olm/clusterserviceversionlist.go [hunk 8]
```diff
@@ -45,3 +46,30 @@ func ListClusterServiceVersion(
 
 	return csvObjects, nil
 }
+
+// ListAllInNamespaceWithNamePattern returns a cluster-wide csv inventory.
```
Comment (trewest) [inline pkg/olm/clusterserviceversionlist.go:50]: Please update the comment to reflect that this function is not cluster-wide and it only return csvs that contain the name pattern

**Review by trewest — CHANGES_REQUESTED** [review]
Just a few comments

File: pkg/olm/clusterserviceversionlist.go [hunk 9]
```diff
@@ -45,3 +46,36 @@ func ListClusterServiceVersion(
 
 	return csvObjects, nil
 }
+
+// ListClusterServiceVersionWithNamePattern returns a cluster-wide csv inventory filtered by the name pattern.
+func ListClusterServiceVersionWithNamePattern(
+	apiClient *clients.Settings,
```
Comment (kononovn) [inline pkg/olm/clusterserviceversionlist.go:52]: Please format augments according convention> https://github.com/openshift-kni/eco-goinfra#functions-format

func Function(
    argInt1 int,
    argInt2 int,
    argInt3 int,
    argInt4 int,
    argString1 string,
    argString2 string,
    argString3 string,
    argString4 string) {
    ...
}

File: pkg/olm/clusterserviceversionlist.go [hunk 10]
```diff
@@ -45,3 +46,36 @@ func ListClusterServiceVersion(
...
+	namePattern, nsname string,
+	options metaV1.ListOptions) ([]*ClusterServiceVersionBuilder, error) {
+	if namePattern == "" {
+		glog.V(100).Info("The namePattern field to filter-out all relevant csv could not be empty")
+
+		return nil, fmt.Errorf("the namePattern field to filter-out all relevant csv could not be empty")
+	}
+
+	glog.V(100).Infof("Listing csv filtered by the name pattern %s are found in %s namespace", namePattern, nsname)
```
Comment (trewest) [inline pkg/olm/clusterserviceversionlist.go:61]:
```suggestion
	glog.V(100).Infof("Listing csv filtered by the name pattern %s in %s namespace", namePattern, nsname)
```

File: pkg/olm/clusterserviceversionlist.go [hunk 11]
```diff
@@ -45,3 +46,36 @@ func ListClusterServiceVersion(
...
 	return csvObjects, nil
 }
+
+// ListClusterServiceVersionWithNamePattern returns a cluster-wide csv inventory filtered by the name pattern.
+func ListClusterServiceVersionWithNamePattern(
+	apiClient *clients.Settings,
+	namePattern, nsname string,
+	options metaV1.ListOptions) ([]*ClusterServiceVersionBuilder, error) {
+	if namePattern == "" {
+		glog.V(100).Info("The namePattern field to filter-out all relevant csv could not be empty")
```
Comment (trewest) [inline pkg/olm/clusterserviceversionlist.go:56]:
```suggestion
		glog.V(100).Info("The namePattern field to filter out all relevant csvs cannot be empty")
```

File: pkg/olm/clusterserviceversionlist.go [hunk 12]
```diff
@@ -45,3 +46,36 @@ func ListClusterServiceVersion(
...
+
+		return nil, fmt.Errorf("the namePattern field to filter-out all relevant csv could not be empty")
```
Comment (trewest) [inline pkg/olm/clusterserviceversionlist.go:58]:
```suggestion
		return nil, fmt.Errorf("the namePattern field to filter out all relevant csvs cannot be empty")
```

File: pkg/olm/clusterserviceversionlist.go [hunk 13]
```diff
@@ -45,3 +46,36 @@ func ListClusterServiceVersion(
...
+	}
+
+	glog.V(100).Infof("Listing csv filtered by the name pattern %s are found in %s namespace", namePattern, nsname)
+
+	notFilteredCsvList, err := ListClusterServiceVersion(apiClient, nsname, options)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to list all clusterserviceversions in %s namespace due to %s",
```
Comment (trewest) [inline pkg/olm/clusterserviceversionlist.go:66]: I think you should standardize on either csvs or clusterserviceversions. Try not to mix them for the sake of consistency.

**Comment by trewest** [comment]
@elenagerman Hey can you replace csv with clusterserviceversion? It looks like the other parts of the package use clusterserviceversion and we want it to be consistent

File: pkg/olm/clusterserviceversionlist.go (hunk 9 above)
Comment (elenagerman) [reply pkg/olm/clusterserviceversionlist.go:52]: I'm sorry, but I don't understant what is wrong with the original version. And, btw, it is identical to the ListClusterServiceVersion

Comment (achuzhoy) [reply pkg/olm/clusterserviceversionlist.go:53]: seems like namePattern, nsname string need to be on separate lines, i.e.

```
namePattern string, 
nsname string
```

File: pkg/olm/clusterserviceversionlist.go [hunk 14]
```diff
@@ -45,3 +46,40 @@ func ListClusterServiceVersion(
...
+			"the namePattern field to filter out all relevant clusterServiceVersion cannot be empty")
+	}
+
+	glog.V(100).Infof("Listing clusterServiceVersion filtered by the name pattern %s in %s namespace",
+		namePattern, nsname)
+
+	notFilteredCsvList, err := ListClusterServiceVersion(apiClient, nsname, options)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to list all clusterserviceversions in %s namespace due to %s",
```
Comment (achuzhoy) [inline pkg/olm/clusterserviceversionlist.go:70]: instead of "%s namespace" should be "namespace %s"

File: pkg/olm/clusterserviceversionlist.go [hunk 15]
```diff
@@ -45,3 +46,40 @@ func ListClusterServiceVersion(
...
+	options metaV1.ListOptions) ([]*ClusterServiceVersionBuilder, error) {
+	if namePattern == "" {
+		glog.V(100).Info(
+			"The namePattern field to filter out all relevant clusterServiceVersion cannot be empty")
+
+		return nil, fmt.Errorf(
+			"the namePattern field to filter out all relevant clusterServiceVersion cannot be empty")
+	}
+
+	glog.V(100).Infof("Listing clusterServiceVersion filtered by the name pattern %s in %s namespace",
```
Comment (achuzhoy) [inline pkg/olm/clusterserviceversionlist.go:64]: %s should come after namespace, i.e. "in namespace %s"

File: pkg/olm/clusterserviceversion.go [hunk 16]
```diff
@@ -120,7 +120,44 @@ func (builder *ClusterServiceVersionBuilder) GetAlmExamples() (string, error) {
...
+
+	if err != nil {
+		return false, fmt.Errorf("failed to get phase value for %s clusterserviceversion in %s namespace due to %w",
+			builder.Definition.Name, builder.Definition.Namespace, err)
+	}
+
+	return phase == "Succeeded", nil
+}
+
+// GetPhase get current clusterServiceVersion phase.
```
Comment (achuzhoy) [inline pkg/olm/clusterserviceversion.go:146]: gets

File: pkg/olm/clusterserviceversion.go [hunk 17]
```diff
@@ -120,7 +120,44 @@ func (builder *ClusterServiceVersionBuilder) GetAlmExamples() (string, error) {
...
+func (builder *ClusterServiceVersionBuilder) GetPhase() (oplmV1alpha1.ClusterServiceVersionPhase, error) {
+	if valid, err := builder.validate(); !valid {
+		return "", err
+	}
+
+	glog.V(100).Infof("Get %s clusterServiceVersion in %s namespace phase",
```
Comment (achuzhoy) [inline pkg/olm/clusterserviceversion.go:152]: The above sentence doesn't look right.
Maybe: 
"Get %s clusterServiceVersion phase in namespace %s"

File: pkg/olm/clusterserviceversion.go [hunk 18]
```diff
@@ -120,7 +120,44 @@ func (builder *ClusterServiceVersionBuilder) GetAlmExamples() (string, error) {
 		}
 	}
 
-	return "", fmt.Errorf("%s not found in given csv named %v", almExamples, builder.Definition.Name)
+	return "", fmt.Errorf("%s not found in given clusterServiceVersion named %v",
+		almExamples, builder.Definition.Name)
+}
+
+// IsSucceeded check if the clusterServiceVersion is Succeeded.
```
Comment (achuzhoy) [inline pkg/olm/clusterserviceversion.go:127]: checks

Comment (achuzhoy) [reply pkg/olm/clusterserviceversion.go:127]: Also, I'd rename the function to isSuccessful for readability and update the comments with is Succeeded to "is Successful"

File: pkg/olm/clusterserviceversionlist.go (hunk 9 above)
Comment (elenagerman) [reply pkg/olm/clusterserviceversionlist.go:53]: thank you Sasha!

**Comment by trewest** [comment]
> @elenagerman Hey can you replace all references to csv with clusterserviceversion? It looks like the other parts of the package use clusterserviceversion and we want it to be consistent

@elenagerman It looks like changed all logs/comments to clusterServiceVersion. Can you changes it to all lowercase (clusterserviceversion) so that it is consistent?

**Review by cdvultur — DISMISSED** [review]
/lgtm

**Review by kononovn — APPROVED** [review]
/lgtm

**Pull request merged**